crates/bevy_ui/src/focus.rs

```diff
@@ -140,23 +120,81 @@ pub fn ui_focus_system(
 
     moused_over_z_sorted_nodes.sort_by_key(|(_, _, _, z)| -*z);
 
-    let mut moused_over_z_sorted_nodes = moused_over_z_sorted_nodes.into_iter();
-    // set Clicked or Hovered on top nodes
-    for (entity, focus_policy, interaction, _) in moused_over_z_sorted_nodes.by_ref() {
-        if let Some(mut interaction) = interaction {
-            if mouse_clicked {
-                // only consider nodes with Interaction "clickable"
-                if *interaction != Interaction::Clicked {
-                    *interaction = Interaction::Clicked;
-                    // if the mouse was simultaneously released, reset this Interaction in the next
-                    // frame
-                    if mouse_released {
-                        state.entities_to_reset.push(entity);
-                    }
-                }
-            } else if *interaction == Interaction::None {
-                *interaction = Interaction::Hovered;
+    set_top_nodes_as_clicked_or_hovered(
+        moused_over_z_sorted_nodes,
+        mouse_button_input,
+        touches_input,
+        state,
+    )
+}
+
+fn contains_cursor(
```

Comment:
This feels like it should be a method on Node or something to that effect?

Reply:
I also thought that this function looks a bit odd as a standalone. But is `Node` the right place as `Node` is actually indifferent to positions?

Reply:
Yeah 🤔 TBH, this feels like functionality that should be pub that we reuse internally, but I can't immediately figure out the best design for that.